mmv1/third_party/terraform/services/certificatemanager/data_source_google_certificate_manager_certificate_map.go

```diff
@@ -0,0 +1,43 @@
+package certificatemanager
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-google/google/tpgresource"
+	transport_tpg "github.com/hashicorp/terraform-provider-google/google/transport"
+)
+
+func DataSourceGoogleCertificateManagerCertificateMap() *schema.Resource {
+
+	dsSchema := tpgresource.DatasourceSchemaFromResourceSchema(ResourceCertificateManagerCertificateMap().Schema)
+	tpgresource.AddRequiredFieldsToSchema(dsSchema, "name")
```

Comment:
Do I need to add `tpgresource.AddOptionalFieldsToSchema(dsSchema, "project")` ? How to know If this is needed or not?

Reply:
I would also add `tpgresource.AddOptionalFieldsToSchema(dsSchema, "project")`. Project is used in the id of the data source. So we will need it there as well

Reply:
Done. Thanks!